### PR TITLE
React Capstone project

### PR DESCRIPTION
**Pull request by Sahar-AbdelSamad**
In this project I:

- Created react app
- Added some dependencies `react-redux`, `react-router-dom`, `redux`,  `redux-devtools-extension`, `redux-logger`, `redux-thunk`
-  Added Linters
- Created directories for Redux files using `ducks pattern`
- Fetched and rendered data from this [Financial modeling prep API](https://financialmodelingprep.com/developer/docs/)
- Added a Home page that lists the companies listed on `Nasdaq-100` and `DJIA` market exchanges
- Added a search bar that filters the companies by their name or symbol
- Added a dropdown listing all the companies
- Added detail page showing data retrieved from the API,  every company has a unique route within the SPA
- Added a `Back` button (<) that navigates to the home page
- Styled the page following these [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs))
- Set up React Testing Library
- Created unit tests for all React components using Jest snapshots
- [Deployed the project](https://sahar-abdelsamad.github.io/company-profile/)
- [Recorded a video presenting the features of the project](https://www.loom.com/share/76bb4e86919b43fd989a4011d07b83dd)

### CODE REVIEW

File: src/__tests__/Company.test.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    .create(
+      <Provider store={store}>
+        <Router>
+          <Company />
+        </Router>
+      </Provider>,
+    )
+    .toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (uwadonat) [inline src/__tests__/Company.test.js:19]: - [x] Please, add more unit tests for several components of the app and some pure functions in Redux code.
- [x] Some of your tests are failing. kindly work on it.
Screenshot from 2022-03-11 00-40-58

**Review by uwadonat — CHANGES_REQUESTED** [review]
Hi **[Sahar-AbdelSamad](https://github.com/Sahar-AbdelSamad),** 

Good job so far!
There are some issues that you still need to work on for your project to be approved but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Your Readme file and PR message are very descriptive ✔️
- Your design looks nice ✔️
- Good presentation ✔️

## Suggested changes

- [x] _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/__tests__/Company.test.js (hunk 1 above)
Comment (Sahar-AbdelSamad) [reply src/__tests__/Company.test.js:19]: Thank you for your review :+1: I have implemented the required changes :+1:

**Review by andersonlebon — APPROVED** [review]
## Hi @Sahar-AbdelSamad 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is  professional :heavy_check_mark: 
- Your design looks nice :heavy_check_mark: 

## Optional suggestions
- [ ] Kindly Check the comments under the review._
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**